### DISCUSSION TITLE
Import workflow and photo names

### DISCUSSION THREAD

**Discussion by jeannotalpin** [photoprism/photoprism#2544] [Q&A]
Hello all,
Big fan of Photoprism, an excellent software :)

I have started using it as a google Photo replacement and I have a question on the "official" workflow to import new photos.

For the moment I work as this:
1. Edit my pictures on my laptop and archive the raw files on a NAS (external to photoprism)
2. Create a new album in Photoprism for the event
3. Upload (via web interface) the selected photos to the newly created album

2 questions:
Most of the the pictures I upload like this have an "unknown/YEAR" name (see picture):
1. How do I fix this now (batch edit?)
2. How is this name determined by Photoprism? Is it based on a tag? Should I do something at the raw processing stage to make sure the name is relevent?

Thanks.

![image](https://user-images.githubusercontent.com/480536/180642592-6b653a80-2339-493a-8d78-2d363c20851b.png)

**Comment by graciousgrey** [comment]
Hello :)

The titles are created based on metadata. In case your files would have titles in their exif/xmp metadata, PhotoPrism would use these titles. Otherwise the title will be created dynamically depending what information is available for a photo e.g.  keywords, labels or locations and the year.
If you add information such as labels or location details to your photos, the title should be updated automatically after a while.

Batch edit is not implemented yet. You could use exiftool to add titles to multiple photos at once.
